src/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx

```diff
@@ -67,8 +67,6 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
       ]}
       ZeroState={ZeroState}
       userPreferredMetric={userPreferences?.metric}
-      // Setting the key here to enforce a remount of the component when changing pages or query parameter filters within the query.
-      key={match.location.pathname + match.location.search}
```

Comment:
I'm not sure if I understand correctly why this causes the flashing images. Is it because forcing the `ArtworkFilterContextProvider` with the `key` prop also remounts it's children?

Reply:
Yeah, key will forcibly re-render the component and anything under it. That means images will be re-loaded.

Reply:
Thanks, then adding a key here is indeed not a good idea.

Reply:
Ah @dzucconi - this was the same issue that fixed things before, I (think). 

Re URL state -- yep, driving by URL state, we've been talking about this for a while. It seems [straight forward](https://github.com/artsy/volt-v2/blob/main/src/pages/demand/hooks/useDemandFilters.tsx) 🤷